src/main/java/org/terasology/module/behaviors/actions/SetTargetToNearbyBlockNode.java

```diff
@@ -27,47 +28,81 @@
                     .getJpsPlugin(actor.getEntity());
             if (locationComponent != null) {
-                Vector3ic target = findRandomNearbyBlock(
-                        Blocks.toBlockPos(locationComponent.getWorldPosition(new Vector3f())),
-                        plugin);
+                Vector3i startBlock = Blocks.toBlockPos(locationComponent.getWorldPosition(new Vector3f()));
+                logger.debug("... [{}]  start position: {}", actor.getEntity().getId(), startBlock);
+                Vector3ic target = findRandomNearbyBlock(startBlock, plugin);
+                if (!plugin.isWalkable(target)) {
+                    logger.debug("... [{}] target position: {} not walkable", actor.getEntity().getId(), target);
+                    return BehaviorState.FAILURE;
+                }
                 moveComponent.target.set(target);
+                moveComponent.setPathGoal(new Vector3i(target));
                 actor.save(moveComponent);
+                logger.debug("... [{}] target position: {} - distance: {}", actor.getEntity().getId(), target, target.sub(startBlock, new Vector3i()).length());
             } else {
+                logger.debug("... [{}] failed", actor.getEntity().getId());
                 return BehaviorState.FAILURE;
             }
         }
         return BehaviorState.SUCCESS;
     }
 
+    /**
+     * TODO: make sure the block is on the ground, i.e., the actor is able to stay at at that location
+     *       obviously, this depends on the entities movement modes, a flying entity can obviously just hover a round...
+     *       => the question now is how a plugin can contribute to the path, but should be excluded from finding a target ...
+     *
+     * TODO: can this be solved using different sets of plugins?
+     *          movementTypes: ["walking", "leaping"]
+     *          idleTypes: ["walking"]
+     *       ideally, we can instantiate all plugins once an re-use them in different contexts (but that might prove difficult due to
+     *       concurrency when building up caches?)
+     *
+     * @param startBlock
+     * @param plugin
+     * @return
+     */
     private Vector3ic findRandomNearbyBlock(Vector3ic startBlock, JPSPlugin plugin) {
         Vector3i currentBlock = new Vector3i(startBlock);
+        long currentDistance = 0;
+        // general distance to startBlock is defined by number of steps (currently at least 3, at max 13)
         for (int i = 0; i < random.nextInt(10) + 3; i++) {
             BlockRegionc neighbors = new BlockRegion(currentBlock).expand(1, 1, 1);
             Vector3ic[] allowedBlocks = Iterators.toArray(
                     Iterators.transform(Iterators.filter(neighbors.iterator(),
-                                    (candidate) -> plugin.isReachable(currentBlock, candidate)),
+                                    (candidate) -> candidate.distanceSquared(startBlock) > currentDistance
```

Comment:
This condition is probably not very good as it easily runs into local dead ends, but it at least prevents us from jumping back and forth while searching... We have a TODO below to improve on this.